nextroute/schema/input.go

```diff
@@ -7,129 +7,217 @@ import (
+	// Vehicles to route.
+	Vehicles []Vehicle `json:"vehicles,omitempty"`
+	// Stops that will be routed and assigned to the vehicles.
+	Stops []Stop `json:"stops,omitempty"`
+	// AlternateStops a set of alternate stops for vehicles.
 	AlternateStops *[]AlternateStop `json:"alternate_stops,omitempty"`
 }
 
 // Defaults contains default values for vehicles and stops.
 type Defaults struct {
+	// Vehicles default values for vehicles.
 	Vehicles *VehicleDefaults `json:"vehicles,omitempty"`
-	Stops    *StopDefaults    `json:"stops,omitempty"`
+	// Stops default values for stops.
+	Stops *StopDefaults `json:"stops,omitempty"`
 }
 
 // VehicleDefaults contains default values for vehicles.
 type VehicleDefaults struct {
-	Capacity                any        `json:"capacity,omitempty"`
-	StartLevel              any        `json:"start_level,omitempty"`
-	StartLocation           *Location  `json:"start_location,omitempty"`
-	EndLocation             *Location  `json:"end_location,omitempty"`
-	Speed                   *float64   `json:"speed,omitempty"`
-	StartTime               *time.Time `json:"start_time,omitempty"`
-	EndTime                 *time.Time `json:"end_time,omitempty"`
-	MinStops                *int       `json:"min_stops,omitempty"`
-	MinStopsPenalty         *float64   `json:"min_stops_penalty,omitempty"`
-	MaxStops                *int       `json:"max_stops,omitempty"`
-	MaxDistance             *int       `json:"max_distance,omitempty"`
-	MaxDuration             *int       `json:"max_duration,omitempty"`
-	MaxWait                 *int       `json:"max_wait,omitempty"`
-	CompatibilityAttributes *[]string  `json:"compatibility_attributes,omitempty"`
-	ActivationPenalty       *int       `json:"activation_penalty,omitempty"`
-	AlternateStops          *[]string  `json:"alternate_stops,omitempty"`
+	// Capacity of the vehicle.
+	Capacity any `json:"capacity,omitempty"`
+	// StartLevel initial level of the vehicle.
+	StartLevel any `json:"start_level,omitempty"`
+	// StartLocation location where the vehicle starts..
+	StartLocation *Location `json:"start_location,omitempty"`
+	// EndLocation location where the vehicle ends..
+	EndLocation *Location `json:"end_location,omitempty"`
+	// Speed of the vehicle in meters per second.
+	Speed *float64 `json:"speed,omitempty"`
+	// StartTime time when the vehicle starts its route.
+	StartTime *time.Time `json:"start_time,omitempty"`
+	// EndTime latest time at which the vehicle ends its route.
+	EndTime *time.Time `json:"end_time,omitempty"`
+	// MinStops minimum stops that a vehicle should visit.
+	MinStops *int `json:"min_stops,omitempty"`
+	// MinStopsPenalty penalty for not visiting the minimum number of stops.
+	MinStopsPenalty *float64 `json:"min_stops_penalty,omitempty"`
+	// MaxStops maximum number of stops that the vehicle can visit.
+	MaxStops *int `json:"max_stops,omitempty"`
+	// MaxDistance maximum distance in meters that the vehicle can travel.
```

Comment:
A short note that this is hard constraint is probably wise, since we don't distinguish hard vs. soft on the field name. This applies for other fields here as well.